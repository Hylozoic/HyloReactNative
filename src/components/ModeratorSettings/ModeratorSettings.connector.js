import { connect } from 'react-redux'
import getCommunity from '../../store/selectors/getCommunity'
import getCurrentCommunityId from '../../store/selectors/getCurrentCommunityId'
import getMe from '../../store/selectors/getMe'

import {
  fetchModeratorSuggestions,
  clearModeratorSuggestions,
  getModerators,
  addModerator,
  removeModerator,
  fetchModerators
} from './ModeratorSettings.store'
import getPerson from '../../store/selectors/getPerson'
import { includes } from 'lodash/fp'

export function mapStateToProps (state, props) {
  const communityId = getCurrentCommunityId(state, props)
  const community = getCommunity(state, { id: communityId })
  const moderators = getModerators(state, { communityId })
  const moderatorIds = moderators.map(m => m.id)
  const moderatorSuggestions = state.ModeratorSettings
    .filter(personId => !includes(personId, moderatorIds))
    .map(personId => getPerson(state, { personId }))
  const currentUser = getMe(state, props)

  return {
    currentUser,
    community,
    moderators,
    moderatorSuggestions
  }
}

export function mapDispatchToProps (dispatch, { navigation }) {
  return {
    addModeratorMaker: (id, communityId) => dispatch(addModerator(id, communityId)),
    removeModeratorMaker: (id, isRemoveFromCommunity, communityId) => dispatch(removeModerator(id, communityId, isRemoveFromCommunity)),
    fetchModeratorsMaker: communitySlug => () => dispatch(fetchModerators(communitySlug)),
    showMember: id => navigation.navigate('MemberProfile', { id }),
    fetchModeratorSuggestionsMaker: (communityId, autocomplete) => dispatch(fetchModeratorSuggestions(communityId, autocomplete)),
    clearModeratorSuggestions: () => dispatch(clearModeratorSuggestions())
  }
}

export const mergeProps = (stateProps, dispatchProps, ownProps) => {
  const communitySlug = stateProps.community && stateProps.community.slug
  const communityId = stateProps.community && stateProps.community.id

  const {
    addModeratorMaker,
    removeModeratorMaker,
    fetchModeratorSuggestionsMaker,
    fetchModeratorsMaker
  } = dispatchProps

  return {
    ...stateProps,
    ...dispatchProps,
    ...ownProps,
    addModerator: (id) => addModeratorMaker(id, communityId),
    removeModerator: (id, isRemoveFromCommunity) => removeModeratorMaker(id, isRemoveFromCommunity, communityId),
    fetchModeratorSuggestions: (autocomplete) => fetchModeratorSuggestionsMaker(communityId, autocomplete),
    fetchModerators: communitySlug ? fetchModeratorsMaker(communitySlug) : () => {}
  }
}

export default connect(mapStateToProps, mapDispatchToProps, mergeProps)
