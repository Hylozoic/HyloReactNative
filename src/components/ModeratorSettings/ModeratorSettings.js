import React, { Component } from 'react'
import { Alert, FlatList, Text, View, TouchableOpacity } from 'react-native'
import { get, isEmpty } from 'lodash/fp'
import header from 'util/header'
import Avatar from '../Avatar'
import Icon from '../Icon'
import LoadingScreen from '../Loading'
// Person picker
import scopedFetchPeopleAutocomplete from '../../store/actions/scopedFetchPeopleAutocomplete'
import scopedGetPeopleAutocomplete from '../../store/selectors/scopedGetPeopleAutocomplete'
import PersonPickerItemRow from '../ItemChooser/PersonPickerItemRow'
import styles from './ModeratorSettings.styles'

export default class ModeratorSettings extends Component {
  static navigationOptions = ({navigation}) => header(navigation, {
    headerBackButton: () => navigation.goBack(),
    title: 'Community Moderators'
  })

  componentDidMount () {
    this.props.fetchModerators()
  }

  componentDidUpdate (prevProps, prevState) {
    if (get('community.slug', prevProps) !== get('community.slug', this.props)) {
      this.props.fetchModerators()
    }
  }

  removeModerator = (id) => {
    Alert.alert(
      'Remove Moderator',
      '',
      [
        {text: 'No', onPress: () => {}, style: 'No'},
        {text: 'Yes', onPress: () => this.props.removeModerator(id, false)}
      ],
      {cancelable: false}
    )
  }

  addModerator = ({ id }) => this.props.addModerator(id)

  isMe = (id) => this.props.currentUser.id === id

  _renderModeratorRow = ({item}) => <ModeratorRow
    moderator={item}
    showMember={this.props.showMember}
    removeModerator={this.isMe(item.id) ? null : this.removeModerator} />

  openPersonPicker = () => {
    const { navigation } = this.props
    const screenTitle = 'Add Moderator'
    navigation.navigate('ItemChooserScreen', {
      screenTitle,
      ItemRowComponent: PersonPickerItemRow,
      pickItem: this.addModerator,
      searchPlaceholder: 'Type here to search for people',
      fetchSearchSuggestions: scopedFetchPeopleAutocomplete(screenTitle),
      getSearchSuggestions: scopedGetPeopleAutocomplete(screenTitle)
    })
  }

  render () {
    const { community, moderators } = this.props

    if (isEmpty(moderators)) {
      return <LoadingScreen />
    }

    return <FlatList
      style={styles.container}
      data={moderators}
      keyboardShouldPersistTaps='always'
      keyExtractor={item => item.id.toString()}
      renderItem={this._renderModeratorRow}
      ListHeaderComponent={<View style={styles.headerContainer}>
        <Text style={styles.headerText}>{community.name}</Text>
        <View style={styles.addNewContainer}>
          <TouchableOpacity onPress={this.openPersonPicker}>
            <Text style={styles.addNewButton}><Icon name='Plus' green /> Add New</Text>
          </TouchableOpacity>
        </View>
      </View>} />
  }
}

export function ModeratorRow ({moderator, showMember, removeModerator}) {
  return <TouchableOpacity style={styles.row} onPress={() => showMember(moderator.id)}>
    {moderator.avatarUrl && <Avatar style={{width: 50}} avatarUrl={moderator.avatarUrl} />}
    <Text style={styles.moderatorName}>{moderator.name}</Text>
    {removeModerator && <TouchableOpacity hitSlop={{top: 15, bottom: 15, left: 10, right: 10}} onPress={() => removeModerator(moderator.id)}>
      <Text style={styles.removeButton}>Remove</Text>
    </TouchableOpacity>}
  </TouchableOpacity>
}
