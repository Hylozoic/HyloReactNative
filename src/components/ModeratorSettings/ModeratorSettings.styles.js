import { caribbeanGreen, white, capeCod40, amaranth, rhino20, white60onCaribbeanGreen } from 'style/colors'

export default {
  container: {
    backgroundColor: white
  },
  headerContainer: {
    alignItems: 'center',
    backgroundColor: white
  },
  headerText: {
    backgroundColor: '#FAFBFC',
    padding: 20,
    width: '90%',
    textAlign: 'center',
    marginTop: 15,
    marginBottom: 10,
    borderRadius: 8,
    fontFamily: 'Circular-Book',
    fontSize: 24
  },
  row: {
    flexDirection: 'row',
    alignItems: 'center',
    paddingLeft: '5%',
    paddingRight: '5%',
    paddingTop: 10,
    paddingBottom: 10,
    backgroundColor: white,
    flex: 1
  },
  removeButton: {
    color: amaranth
  },
  addNewContainer: {
    width: '100%',
    paddingLeft: '5%',
    paddingRight: '5%',
    marginTop: 10
  },
  addNewButton: {
    padding: 10,
    fontSize: 18,
    borderStyle: 'dashed',
    borderColor: white60onCaribbeanGreen,
    borderWidth: 2,
    fontFamily: 'Circular-Bold',
    color: caribbeanGreen,
    textAlign: 'center'
  },
  addModeratorContainer: {
    marginTop: 10,
    marginLeft: '5%',
    marginRight: '5%'
  },
  addModeratorButtonsContainer: {
    flexDirection: 'row'
  },
  button: {
    paddingLeft: 15,
    paddingTop: 10
  },
  cancelButton: {
    fontFamily: 'Circular-Bold',
    color: capeCod40
  },
  addButton: {
    fontFamily: 'Circular-Bold',
    color: caribbeanGreen
  },
  autocomplete: {
    flex: 1
  },
  autocompleteContainer: {
    borderWidth: 0,
    flex: 1
  },
  autocompleteInput: {
    borderWidth: 0,
    flex: 1
  },
  autocompleteItem: {
    flexDirection: 'row',
    alignItems: 'center',
    padding: 10
  }
}
