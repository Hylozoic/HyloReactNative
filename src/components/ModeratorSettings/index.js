import { withNavigationFocus } from 'react-navigation-is-focused-hoc'

import component from './ModeratorSettings'
import connector from './ModeratorSettings.connector'

export default withNavigationFocus(connector(component))
