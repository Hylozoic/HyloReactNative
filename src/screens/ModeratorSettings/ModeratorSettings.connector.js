import { connect } from 'react-redux'
import getGroup from 'store/selectors/getGroup'
import getCurrentGroupId from 'store/selectors/getCurrentGroupId'
import getMe from 'store/selectors/getMe'

import {
  fetchModeratorSuggestions,
  clearModeratorSuggestions,
  getModerators,
  addModerator,
  removeModerator,
  fetchModerators
} from './ModeratorSettings.store'
import getPerson from 'store/selectors/getPerson'
import { includes } from 'lodash/fp'

export function mapStateToProps (state, props) {
  const groupId = getCurrentGroupId(state, props)
  const group = getGroup(state, { id: groupId })
  const moderators = getModerators(state, { groupId })
  const moderatorIds = moderators.map(m => m.id)
  const moderatorSuggestions = state.ModeratorSettings
    .filter(personId => !includes(personId, moderatorIds))
    .map(personId => getPerson(state, { personId }))
  const currentUser = getMe(state, props)

  return {
    currentUser,
    group,
    moderators,
    moderatorSuggestions
  }
}

export function mapDispatchToProps (dispatch, { navigation }) {
  return {
    addModeratorMaker: (id, groupId) => dispatch(addModerator(id, groupId)),
    removeModeratorMaker: (id, isRemoveFromGroup, groupId) => dispatch(removeModerator(id, groupId, isRemoveFromGroup)),
    fetchModeratorsMaker: groupSlug => () => dispatch(fetchModerators(groupSlug)),
    showMember: id => navigation.navigate('Member', { id }),
    fetchModeratorSuggestionsMaker: (groupId, autocomplete) => dispatch(fetchModeratorSuggestions(groupId, autocomplete)),
    clearModeratorSuggestions: () => dispatch(clearModeratorSuggestions())
  }
}

export const mergeProps = (stateProps, dispatchProps, ownProps) => {
  const groupSlug = stateProps.group && stateProps.group.slug
  const groupId = stateProps.group && stateProps.group.id

  const {
    addModeratorMaker,
    removeModeratorMaker,
    fetchModeratorSuggestionsMaker,
    fetchModeratorsMaker
  } = dispatchProps

  return {
    ...stateProps,
    ...dispatchProps,
    ...ownProps,
    addModerator: (id) => addModeratorMaker(id, groupId),
    removeModerator: (id, isRemoveFromGroup) => removeModeratorMaker(id, isRemoveFromGroup, groupId),
    fetchModeratorSuggestions: (autocomplete) => fetchModeratorSuggestionsMaker(groupId, autocomplete),
    fetchModerators: groupSlug ? fetchModeratorsMaker(groupSlug) : () => {}
  }
}

export default connect(mapStateToProps, mapDispatchToProps, mergeProps)
