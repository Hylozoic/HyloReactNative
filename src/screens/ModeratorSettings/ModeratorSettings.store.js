import { createSelector as ormCreateSelector } from 'redux-orm'
import orm from 'store/models'

export const MODULE_NAME = 'ModeratorSettings'

// Constants
export const ADD_MODERATOR = `${MODULE_NAME}/ADD_MODERATOR`
export const ADD_MODERATOR_PENDING = `${ADD_MODERATOR}_PENDING`
export const REMOVE_MODERATOR = `${MODULE_NAME}/REMOVE_MODERATOR`
export const REMOVE_MODERATOR_PENDING = `${REMOVE_MODERATOR}_PENDING`
export const FETCH_MODERATOR_SUGGESTIONS = `${MODULE_NAME}/FETCH_MODERATOR_SUGGESTIONS`
export const CLEAR_MODERATOR_SUGGESTIONS = `${MODULE_NAME}/CLEAR_MODERATOR_SUGGESTIONS`
export const FETCH_MODERATORS = `${MODULE_NAME}/FETCH_MODERATORS`

const defaultState = []

export function ormSessionReducer (session, { type, meta, payload }) {
  const { Community, Person } = session
  let community, person

  switch (type) {
    case REMOVE_MODERATOR_PENDING:
      community = Community.withId(meta.communityId)
      const moderators = community.moderators.filter(m =>
        m.id !== meta.personId)
        .toModelArray()
      community.update({ moderators })
      break

    case ADD_MODERATOR_PENDING:
      person = Person.withId(meta.personId)
      Community.withId(meta.communityId).updateAppending({ moderators: [person] })
      break
  }
}

export default function reducer (state = defaultState, action) {
  const { error, type, payload } = action
  if (error) return state

  switch (type) {
    case FETCH_MODERATOR_SUGGESTIONS:
      return payload.data.community.members.items.map(m => m.id)
    case CLEAR_MODERATOR_SUGGESTIONS:
      return []
    default:
      return state
  }
}

export function fetchModerators (slug) {
  return {
    type: FETCH_MODERATORS,
    graphql: {
      query: `query ($slug: String) {
        community (slug: $slug) {
          id
          name
          slug
          moderators (first: 100) {
            hasMore
            items {
              id
              name
              avatarUrl
            }
          }
        }
      }`,
      variables: {
        slug
      }
    },
    meta: {
      extractModel: 'Community'
    }
  }
}

export function fetchModeratorSuggestions (id, autocomplete) {
  return {
    type: FETCH_MODERATOR_SUGGESTIONS,
    graphql: {
      query: `query ($id: ID, $autocomplete: String) {
        community (id: $id) {
          id
          members (first: 10, autocomplete: $autocomplete) {
            hasMore
            items {
              id
              name
              avatarUrl
            }
          }
        }
      }`,
      variables: {
        id, autocomplete
      }
    },
    meta: {
      extractModel: 'Community'
    }
  }
}

export function clearModeratorSuggestions () {
  return {
    type: CLEAR_MODERATOR_SUGGESTIONS
  }
}

export function addModerator (personId, communityId) {
  return {
    type: ADD_MODERATOR,
    graphql: {
      query: `mutation ($personId: ID, $communityId: ID) {
        addModerator(personId: $personId, communityId: $communityId) {
          id
          moderators (first: 100) {
            items {
              id
              name
              avatarUrl
            }
          }
        }
      }`,
      variables: { personId, communityId }
    },
    meta: {
      personId,
      communityId,
      optimistic: true
    }
  }
}

export function removeModerator (personId, communityId, isRemoveFromCommunity) {
  return {
    type: REMOVE_MODERATOR,
    graphql: {
      query: `mutation ($personId: ID, $communityId: ID, $isRemoveFromCommunity: Boolean) {
        removeModerator(personId: $personId, communityId: $communityId, isRemoveFromCommunity: $isRemoveFromCommunity) {
          id
          moderators (first: 100) {
            items {
              id
              name
              avatarUrl
            }
          }
        }
      }`,
      variables: { personId, communityId, isRemoveFromCommunity }
    },
    meta: {
      personId,
      communityId,
      isRemoveFromCommunity,
      optimistic: true
    }
  }
}

// expects props to be of the form {communityId}
export const getModerators = ormCreateSelector(
  orm,
  state => state.orm,
  (state, props) => props.communityId,
  ({ Community }, id) => {
    const community = Community.safeGet({ id })
    if (!community) return []
    return community.moderators.toModelArray()
  }
)
