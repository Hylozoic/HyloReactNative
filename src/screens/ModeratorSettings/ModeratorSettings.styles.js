import { StyleSheet } from 'react-native'

import { caribbeanGreen, white, capeCod, capeCod05, capeCod40, amaranth, rhino20, white60onCaribbeanGreen } from 'style/colors'

export default StyleSheet.create({
  container: {
    backgroundColor: white
  },
  headerContainer: {
    alignItems: 'center',
    backgroundColor: white
  },
  headerText: {
    backgroundColor: capeCod05,
    padding: 5,
    marginTop: 15,
    marginBottom: 10,
    width: '90%',
    textAlign: 'center',
    borderRadius: 4,
    color: capeCod,
    fontFamily: 'Circular-Bold',
    fontSize: 24
  },
  row: {
    width: '90%',
    flexDirection: 'row',
    alignItems: 'center',
    paddingTop: 10,
    paddingBottom: 10,
    marginHorizontal: '5%',
    backgroundColor: white,
    flex: 1,
    borderBottomWidth: StyleSheet.hairlineWidth,
    borderBottomColor: rhino20
  },
  removeButton: {
    color: amaranth,
    fontFamily: 'Circular-Book',
    fontSize: 16
  },
  addNewContainer: {
    width: '90%',
    paddingVertical: 20,
    padding: 5
  },
  addNewButton: {
    padding: 5,
    fontSize: 18,
    borderStyle: 'dashed',
    borderWidth: 1,
    borderColor: caribbeanGreen,
    borderRadius: 4,
    fontFamily: 'Circular-Book',
    color: caribbeanGreen,
    textAlign: 'center'
  },
  cancelButton: {
    fontFamily: 'Circular-Bold',
    color: capeCod40
  },
  addButton: {
    fontFamily: 'Circular-Bold',
    color: caribbeanGreen
  },
  autocomplete: {
    flex: 1
  },
  autocompleteContainer: {
    borderWidth: 0,
    flex: 1
  },
  autocompleteInput: {
    borderWidth: 0,
    flex: 1
  },
  autocompleteItem: {
    flexDirection: 'row',
    alignItems: 'center',
    padding: 10
  },
  moderatorName: {
    flex: 1,
    fontFamily: 'Circular-Bold',
    fontSize: 18
  }
})
