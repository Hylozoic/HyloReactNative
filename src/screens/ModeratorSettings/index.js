import component from './ModeratorSettings'
import connector from './ModeratorSettings.connector'

export default connector(component)
