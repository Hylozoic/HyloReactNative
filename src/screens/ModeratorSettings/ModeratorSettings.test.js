import React from 'react'
import { Alert } from 'react-native'
import ReactShallowRenderer from 'react-test-renderer/shallow'
import { ModeratorSettings, ModeratorRow } from './ModeratorSettings'
import ReactTestRenderer from 'react-test-renderer'

jest.mock('react-native/Libraries/Alert/Alert', () => ({
  alert: jest.fn()
}))

describe('ModeratorSettings', () => {
  const props = {
    moderators: [{ id: 10, name: 'John Current', avatarUrl: 'http://aurl.com' }, { id: 11, name: 'John Other', avatarUrl: 'http://aurl.com' }],
    currentUser: { id: 10 },
    group: { name: 'Group Time' },
    fetchModerators: jest.fn(),
    addModerator: jest.fn(),
    moderatorSuggestions: [],
    fetchModeratorSuggestions: jest.fn(),
    clearModeratorSuggestions: jest.fn()
  }

  it('matches the last snapshot', () => {
    const renderer = new ReactShallowRenderer()

    renderer.render(<ModeratorSettings {...props} />)
    const actual = renderer.getRenderOutput()

    expect(actual).toMatchSnapshot()
  })

  it('fetches molderators on update', () => {
    const props = {
      group: { slug: 'foo' },
      fetchModerators: jest.fn()
    }
    const prevPropsSameSlug = {
      group: { slug: 'foo' },
      fetchModerators: jest.fn()
    }

    const prevPropsDifferentSlug = {
      group: { slug: 'bar' },
      fetchModerators: jest.fn()
    }

    const instance = ReactTestRenderer.create(<ModeratorSettings {...props} />).getInstance()

    expect(props.fetchModerators).toHaveBeenCalledTimes(1)
    instance.componentDidUpdate(prevPropsSameSlug)
    expect(props.fetchModerators).toHaveBeenCalledTimes(1)
    instance.componentDidUpdate(prevPropsDifferentSlug)
    expect(props.fetchModerators).toHaveBeenCalledTimes(2)
  })

  it('renders ModeratorRow', () => {
    const renderer = new ReactShallowRenderer()
    const props = {
      moderator: { id: 10, name: 'John Current', avatarUrl: 'http://aurl.com' },
      showMember: jest.fn(),
      removeModerator: jest.fn()
    }
    renderer.render(<ModeratorRow {...props} />)
    const actual = renderer.getRenderOutput()

    expect(actual).toMatchSnapshot()
  })

  it('renders ModeratorRow without removeModerator', () => {
    const renderer = new ReactShallowRenderer()
    const props = {
      moderator: { id: 10, name: 'John Current', avatarUrl: 'http://aurl.com' },
      showMember: jest.fn(),
      removeModerator: null
    }
    renderer.render(<ModeratorRow {...props} />)
    const actual = renderer.getRenderOutput()

    expect(actual).toMatchSnapshot()
  })

  it('sends alert when removing moderator', () => {
    const instance = ReactTestRenderer.create(<ModeratorSettings {...props} />).getInstance()

    instance.removeModerator(10)
    expect(Alert.alert).toHaveBeenCalled()
  })

  it('addModerator', () => {
    const instance = ReactTestRenderer.create(<ModeratorSettings {...props} />).getInstance()
    const moderatorId = 12
    instance.addModerator({ id: moderatorId })
    expect(props.addModerator).toHaveBeenCalledWith(moderatorId)
  })

  it('shows the Search component appropriately', () => {
    const renderer = ReactTestRenderer.create(<ModeratorSettings {...props} />)
    const instance = renderer.getInstance()
    instance.setState({
      showPicker: true
    })
    expect(renderer.toJSON()).toMatchSnapshot()
  })
})
