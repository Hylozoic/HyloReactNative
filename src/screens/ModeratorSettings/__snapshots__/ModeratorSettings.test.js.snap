// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`ModeratorSettings matches the last snapshot 1`] = `
<FlatList
  ListHeaderComponent={
    <View>
      <View>
        <TouchableOpacity
          onPress={[Function]}
        >
          <Text>
            <Icon
              green={true}
              name="Plus"
            />
             Add New
          </Text>
        </TouchableOpacity>
      </View>
    </View>
  }
  data={
    Array [
      Object {
        "avatarUrl": "http://aurl.com",
        "id": 10,
        "name": "John Current",
      },
      Object {
        "avatarUrl": "http://aurl.com",
        "id": 11,
        "name": "John Other",
      },
    ]
  }
  keyExtractor={[Function]}
  keyboardShouldPersistTaps="always"
  numColumns={1}
  removeClippedSubviews={false}
  renderItem={[Function]}
/>
`;

exports[`ModeratorSettings renders ModeratorRow 1`] = `
<TouchableOpacity
  onPress={[Function]}
>
  <Avatar
    avatarUrl="http://aurl.com"
  />
  <Text>
    John Current
  </Text>
  <TouchableOpacity
    hitSlop={
      Object {
        "bottom": 15,
        "left": 10,
        "right": 10,
        "top": 15,
      }
    }
    onPress={[Function]}
  >
    <Text>
      Remove
    </Text>
  </TouchableOpacity>
</TouchableOpacity>
`;

exports[`ModeratorSettings renders ModeratorRow without removeModerator 1`] = `
<TouchableOpacity
  onPress={[Function]}
>
  <Avatar
    avatarUrl="http://aurl.com"
  />
  <Text>
    John Current
  </Text>
</TouchableOpacity>
`;

exports[`ModeratorSettings shows the Search component appropriately 1`] = `
<RCTScrollView
  ListHeaderComponent={
    <View>
      <View>
        <TouchableOpacity
          onPress={[Function]}
        >
          <Text>
            <Icon
              green={true}
              name="Plus"
            />
             Add New
          </Text>
        </TouchableOpacity>
      </View>
    </View>
  }
  data={
    Array [
      Object {
        "avatarUrl": "http://aurl.com",
        "id": 10,
        "name": "John Current",
      },
      Object {
        "avatarUrl": "http://aurl.com",
        "id": 11,
        "name": "John Other",
      },
    ]
  }
  getItem={[Function]}
  getItemCount={[Function]}
  keyExtractor={[Function]}
  keyboardShouldPersistTaps="always"
  onContentSizeChange={[Function]}
  onLayout={[Function]}
  onMomentumScrollBegin={[Function]}
  onMomentumScrollEnd={[Function]}
  onScroll={[Function]}
  onScrollBeginDrag={[Function]}
  onScrollEndDrag={[Function]}
  removeClippedSubviews={false}
  renderItem={[Function]}
  scrollEventThrottle={50}
  stickyHeaderIndices={Array []}
  viewabilityConfigCallbackPairs={Array []}
>
  <View>
    <View
      onLayout={[Function]}
    >
      <View>
        <View>
          <View
            accessible={true}
            focusable={true}
            onClick={[Function]}
            onResponderGrant={[Function]}
            onResponderMove={[Function]}
            onResponderRelease={[Function]}
            onResponderTerminate={[Function]}
            onResponderTerminationRequest={[Function]}
            onStartShouldSetResponder={[Function]}
          >
            <Text>
              <Icon
                green={true}
                name="Plus"
              />
               Add New
            </Text>
          </View>
        </View>
      </View>
    </View>
    <View
      onLayout={[Function]}
    >
      <View
        accessible={true}
        focusable={true}
        onClick={[Function]}
        onResponderGrant={[Function]}
        onResponderMove={[Function]}
        onResponderRelease={[Function]}
        onResponderTerminate={[Function]}
        onResponderTerminationRequest={[Function]}
        onStartShouldSetResponder={[Function]}
      >
        <View>
          <Image
            source={
              Object {
                "uri": "http://aurl.com",
              }
            }
          />
        </View>
        <Text>
          John Current
        </Text>
      </View>
    </View>
    <View
      onLayout={[Function]}
    >
      <View
        accessible={true}
        focusable={true}
        onClick={[Function]}
        onResponderGrant={[Function]}
        onResponderMove={[Function]}
        onResponderRelease={[Function]}
        onResponderTerminate={[Function]}
        onResponderTerminationRequest={[Function]}
        onStartShouldSetResponder={[Function]}
      >
        <View>
          <Image
            source={
              Object {
                "uri": "http://aurl.com",
              }
            }
          />
        </View>
        <Text>
          John Other
        </Text>
        <View
          accessible={true}
          focusable={true}
          hitSlop={
            Object {
              "bottom": 15,
              "left": 10,
              "right": 10,
              "top": 15,
            }
          }
          onClick={[Function]}
          onResponderGrant={[Function]}
          onResponderMove={[Function]}
          onResponderRelease={[Function]}
          onResponderTerminate={[Function]}
          onResponderTerminationRequest={[Function]}
          onStartShouldSetResponder={[Function]}
        >
          <Text>
            Remove
          </Text>
        </View>
      </View>
    </View>
  </View>
</RCTScrollView>
`;
